lib/puppet/functions/run_script.rb

```diff
@@ -46,7 +46,7 @@ def run_script(scope, script, *nodes)
       Puppet::Pops::Types::ExecutionResult::EMPTY_RESULT
     else
       Puppet::Pops::Types::ExecutionResult.from_bolt(
-        executor.run_script(executor.from_uris(hosts), found)
+        executor.run_script(executor.from_uris(hosts), found, [])
```

Comment:
@joshcooper has mentioned an alternative of modifying the signature in Bolt `run_script` at https://github.com/puppetlabs/bolt/blob/332513f5506a891214ece22a10c45d941cb28940/lib/bolt/executor.rb#L71 to default `arguments = []` ... if that's the case, we should likely modify `run_task` with the same defaults as well.

Reply:
While I can merge this, I can also just fix the impl so that it actually supports arguments. If so, what are they? Just positional values? Can they be anything except strings?

Reply:
:+1: to fixing. The args are positional and always strings.